File: src/main/java/io/junseok/mealmate/domain/member/entity/Member.java
package io.junseok.mealmate.domain.member.entity;

import static javax.persistence.EnumType.STRING;

import io.junseok.mealmate.domain.base.BaseTimeEntity;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "member")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Builder
public class Member extends BaseTimeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id")
    private Long memberId;

    @Column(name = "email")
    private String email;

    @Column(name = "password")
    private String password;

    @Column(name = "activated")
    private boolean activated;

    @Enumerated(STRING)
    private Authority authority;
}

File: src/main/java/io/junseok/mealmate/domain/member/entity/Member.kt
package io.junseok.mealmate.domain.member.entity;


import io.junseok.mealmate.domain.base.BaseTimeEntity
import javax.persistence.*

@Entity
@Table(name = "member")
class Member(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id")
    var memberId: Long? = null,

    @Column(name = "email")
    var email: String,

    @Column(name = "password")
    var password: String,

    @Column(name = "nickname")
    var nickname: String,

    @Column(name = "school")
    var school: String,

    @Column(name = "student_number")
    val studentNumber: String,

    @Column(name = "department")
    var department: String,

    @Column(name = "activated")
    var activated: Boolean,

    @Enumerated(EnumType.STRING)
    val authority: Authority
) : BaseTimeEntity() {

    fun updatePassword(password: String): Unit {
        this.password = password
    }
}